orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteStageHandler.kt

```diff
@@ -78,7 +82,13 @@ class CompleteStageHandler(
               status = SKIPPED
             }
           } else if (status.isFailure) {
-            if (stage.planOnFailureStages()) {
+            var hasOnFailureStages = false
+
+            stage.withAuth {
+              hasOnFailureStages = stage.planOnFailureStages()
```

Comment:
I got some kotlin error when I nested all this within `stage.withAuth` ... something about `return@withStage` not being legitimate.

Broke this apart to get around it.

Reply:
I always think that `return@something` syntax is a good sign that things need breaking apart so I'm 100% down with that.

Reply:
reading through https://kotlinlang.org/docs/reference/returns.html and not understanding why you can't just do `return@withStage` here...  @robfletcher, if you know, would love to understand it.